### PR TITLE
Fix glitch in timer when loading a sale.

### PR DESCRIPTION
**Pull request by mprasanjith**
## Description

- Avoid passing the sale `endDate` as the initial value to render to the timer.
- Extract the timer tick to a separate function.
- Trigger the tick function before setting up `setInterval`, to set the initial value.

## Motivation and Context

#207 

## How Has This Been Tested?

Tested locally using start-xdai.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rossneilson** [comment]
Same with the other PR you need to check the github action that runs on every PR to ensure the build works fine.
Currently has unused values that eslint does not like.

**Comment by mprasanjith** [comment]
@rossneilson yeah fixed both of them now. Running yarn lint locally spits out a bunch of unused variable warnings. Will make a PR cleaning those up.

**Review by martinkrung — COMMENTED** [review]
good for me, but @hhamud should check this too, because he worked most on this

File: src/views/Sale/components/HeaderItem/index.tsx [hunk 1]
```diff
@@ -77,15 +77,20 @@ export function HeaderItem({
   const [, setTime] = useState<number>(0)
   const [descriptionText, setDescriptionText] = useState<string>(description)
 
+  const runTimer = () => {
```
Comment (hhamud) [inline src/views/Sale/components/HeaderItem/index.tsx:80]: Good work, nice and clean.

**Review by hhamud — APPROVED** [review]
I've checked out the branch and it looks good on my end. However, next time we should add unit tests for each change that we do but time is short atm.

**Pull request merged**